src/activity_handlers.cpp

```diff
@@ -1501,6 +1502,51 @@ void activity_handlers::repair_item_finish( player_activity *act, player *p )
     act->moves_left = actor->move_cost;
 }
 
+void activity_handlers::mend_item_finish( player_activity *act, player *p )
+{
+    item_location target;
+    switch( static_cast<item_location::type>( act->index ) ) {
```

Comment:
Ok. Can you open an issue after merging and I'll start work on serialization of `item_location` next as that would have a lot of extra benefits